### PR TITLE
refactor(schemas): infer type in itemlogin service

### PR DESCRIPTION
**Pull request by Defferrard**

### CODE REVIEW

File: src/services/itemLogin/schemas.ts [hunk 1]
```diff
@@ -1,83 +1,88 @@
...
-  .prop('item', item)
-  .prop('createdAt', S.string())
-  .prop('updatedAt', S.string())
-  .prop('id', uuid);
+export const credentialsSchemaRef = registerSchemaAsRef(
+  'itemLoginCredentials',
+  'Item Login Credentials',
+  Type.Object(
+    {
+      username: Type.String({ minLength: 3, maxLength: 50, pattern: '^\\S+( \\S+)*$' }),
```
Comment (ReidyT) [inline src/services/itemLogin/schemas.ts:17]: I don't know if we want to extract the pattern in the same const for the username and password for allowing their to evolve independently, but should we extract the patterns in varioble or other, to simplify the readability of the Regex?

File: src/services/itemLogin/schemas.ts [hunk 2]
```diff
@@ -1,83 +1,88 @@
...
+  'Item Login Schema',
+  itemLoginSchemaSchema,
+);
 
 export const login = {
-  params: idParam,
-  querystring: S.object().additionalProperties(false).prop('m', S.boolean()),
-  body: credentials,
+  params: entityIdSchemaRef,
+  body: credentialsSchemaRef,
```
Comment (ReidyT) [inline src/services/itemLogin/schemas.ts:45]: What happens to the query string?

**Review by ReidyT — APPROVED** [review]
LGTM

**Review by pyphilia — APPROVED** [review]
LGTM 💪

File: src/services/itemLogin/schemas.ts (hunk 2 above)
Comment (Defferrard) [reply src/services/itemLogin/schemas.ts:45]: It is not used, so I felt free to remove it!

File: src/services/itemLogin/schemas.ts (hunk 1 above)
Comment (Defferrard) [reply src/services/itemLogin/schemas.ts:17]: It'll use the customType username in a future PR 👍🏻

**Pull request merged**